torch/_dynamo/guards.py

```diff
@@ -493,6 +494,15 @@ def DETERMINISTIC_ALGORITHMS(self, guard: Guard):
             code = "not ___are_deterministic_algorithms_enabled()"
         self._produce_guard_code(guard, [code])
 
+    def TORCH_FUNCTION_STATE(self, guard: Guard):
+        assert guard.source is GuardSource.GLOBAL
+        code = None
+        if convert_frame.initial_torch_function_state:
+            code = "___is_torch_function_enabled()"
+        else:
+            code = "not ___is_torch_function_enabled()"
```

Comment:
nit: 1 liner like `code = ("not " if not convert_frame.initial_torch_function_state else "") + "___is_torch_function_enabled()"` or something like it.